shell/common/shell.cc

```diff
@@ -1223,7 +1223,20 @@ void Shell::OnEngineHandlePlatformMessage(
   }
 
   if (platform_message_handler_) {
-    if (route_messages_through_platform_thread_) {
+    if (route_messages_through_platform_thread_ &&
+        !platform_message_handler_
+             ->DoesHandlePlatformMessageOnPlatformThread()) {
+#if _WIN32
+      // On Windows capturing a TaskRunner with a TaskRunner will cause an
+      // uncaught exception in process shutdown because of the deletion order of
+      // global variables. See also
+      // https://github.com/flutter/flutter/issues/111575.
+      // This won't be an issue until Windows supports background platform
+      // channels. Then this can potentially be addressed by capturing a
+      // weak_ptr to an object that retains the ui TaskRunner, instead of the
+      // TaskRunner directly.
```

Comment:
File an issue?

Reply:
(Or link to an existing one for supporting background channels on Windows, with context added there)

Reply:
done